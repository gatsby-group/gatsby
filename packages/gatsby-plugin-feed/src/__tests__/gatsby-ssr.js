const { onRenderBody } = require(`../gatsby-ssr`)

beforeEach(() => {
  global.__PATH_PREFIX__ = ``
})

describe(`Adds <Link> for feed to head`, () => {
  it(`creates Link if feeds does exist`, async () => {
    const pluginOptions = {
      feeds: [
        {
          output: `feed.xml`,
        },
      ],
    }
    const setHeadComponents = jest.fn()

    await onRenderBody(
      {
        setHeadComponents,
      },
      pluginOptions
    )

    expect(setHeadComponents).toMatchSnapshot()
    expect(setHeadComponents).toHaveBeenCalledTimes(1)
  })
  it(`creates multiple Link if feeds are several`, async () => {
    const pluginOptions = {
      feeds: [
        {
          output: `gryffindor/feed.xml`,
        },
        {
          output: `ravenclaw/feed.xml`,
        },
      ],
    }
    const setHeadComponents = jest.fn()

    await onRenderBody(
      {
        setHeadComponents,
      },
      pluginOptions
    )

    expect(setHeadComponents).toMatchSnapshot()
    expect(setHeadComponents).toHaveBeenCalledTimes(1)
  })
})
