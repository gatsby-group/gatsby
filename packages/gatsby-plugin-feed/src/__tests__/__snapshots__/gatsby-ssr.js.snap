// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds <Link> for feed to head creates Link if feeds does exist 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        <link
          href="/feed.xml"
          rel="alternate"
          type="application/rss+xml"
        />,
      ],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`Adds <Link> for feed to head creates multiple Link if feeds are several 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        <link
          href="/gryffindor/feed.xml"
          rel="alternate"
          type="application/rss+xml"
        />,
        <link
          href="/ravenclaw/feed.xml"
          rel="alternate"
          type="application/rss+xml"
        />,
      ],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
